sssd_test_framework/utils/sssd.py

```diff
@@ -132,6 +133,8 @@ def async_start(
         :return: Running SSH process.
         :rtype: Process
         """
+        self.set_service_user(service_user)
```

Comment:
Can we skip calling `set_service_user` method when `service_user` argument (taken from default value, or passed as argument) == default service_user value 'sssd' ?

Reply:
We could query current value (I wouldn't rely of "default").
Do you think it makes sense?

Reply:
I mean, this is an ssh (or podman) command anyway.

Reply:
> Do you think it makes sense?

Makes sense to me, I was trying to avoid calling `set_service_user` every single time these `*start` or `start` methods are run.

Reply:
This check (if current user matches requested) would be within `set_service_user()` anyway.
And to perform this check, we need to execute ssh/podmand command on a host anyway.
So in best case it will be the same 1 command, in worst case it will be 2 commands (check + change) instead of 1.
It can also save backing up a file.
Does it really worth the trouble?

Reply:
`start` method is used quite often so it seems worthwhile IMO especially the `self.host.conn.run("systemctl daemon-reload")` seems intensive.

```
justin@fedora:~/github/sssd/src/tests/system$ grep -irn 'sssd.start' tests/ | wc -l
190
```

@pbrezina Can we read the SSSD service user at host startup and check this value to conditionally run set_service_user() or do you prefer to keep the current PR state?

Reply:
Ok, I'll add a check at the beginning of `set_service_user()` but I wouldn't rely on a state at "host startup" (what moment is it?)
What if host starts with sssd-2.9 and sssd-2.10 is installed later?
What is cleanup from previous test didn't work as expected and didn't restore "default" sssd.service (or is this out of scope)?

Reply:
Maybe it makes sense to hold off on this check until all supported SSSD versions where test framework are running with the same default service user 'sssd', and we won't be switching back and forth often with

`@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))`

Reply:
For sssd < 2.10 `set_service_user()` is a no-op because of:
```
            if not self.host.features["non-privileged"]:
                return
```

I didn't get "we won't be switching back and forth often". Could you please clarify?

The idea of `@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))` is to run the same test twice with SSSD running both under 'root' and 'sssd'. This should be small set of selected core tests.

Reply:
My proposal was to:
- read the `User=$user` value from `/usr/lib/systemd/system/sssd.service` at the start of the pytest session, and will last the entire pytest session. I think this can be done adding functionality to https://github.com/SSSD/sssd-test-framework/blob/master/sssd_test_framework/hosts/client.py but I need to test it.
- add conditional check, checking against the above value :
```
    if service_user == self.host.default_sssd_service_user:
                return
```
After each test run, the test restores the original `/usr/lib/systemd/system/sssd.service`.
>  What if host starts with sssd-2.9 and sssd-2.10 is installed later?

Does this happen now for some system tests? I didn't think so but i'm not sure.

> What if cleanup from previous test didn't work as expected and didn't restore "default" sssd.service (or is this out of scope)?

If cleanup doesn't work properly then test environment is tainted and all bets are off, we have seen this already in certain test failures in PRCI.

Reply:
> > What if host starts with sssd-2.9 and sssd-2.10 is installed later?
> 
> Does this happen now for some system tests? I didn't think so but i'm not sure.

I was thinking of a case where Fedora comes with stock sssd-2.9 and CI installs sssd-2.10 on top.
But by the time of pytest session it should be already upgraded, so not an issue.
I wonder if we ever want to introduce a real "upgrade test" in the CI...

Ok, let's see if we can cache it.

Reply:
> `start` method is used quite often so it seems worthwhile IMO especially the `self.host.conn.run("systemctl daemon-reload")` seems intensive.
> 
> ```
> justin@fedora:~/github/sssd/src/tests/system$ grep -irn 'sssd.start' tests/ | wc -l
> 190
> ```
> 
> @pbrezina Can we read the SSSD service user at host startup and check this value to conditionally run set_service_user() or do you prefer to keep the current PR state?

Yes, you can add it to `pytest_setup` method. This is executed only once and SSSD version under test is already installed.

Reply:
@justin-stephenson, please take a look at the latest version.